tensorrt_llm/_torch/auto_deploy/shim/ad_executor.py

```diff
@@ -112,6 +116,90 @@ def calculate_max_num_blocks(
         return self.num_blocks, 0
 
 
+class ADHiddenStateManager(Eagle3ResourceManager):
```

Comment:
can this class be used for Eagle3 only or other MTP approaches as well that require hidden states? How much effort would it be to extend it